unified-runtime/source/adapters/level_zero/platform.cpp

```diff
@@ -494,6 +494,30 @@ ur_result_t ur_platform_handle_t_::initialize() {
   return UR_RESULT_SUCCESS;
 }
 
+bool ur_platform_handle_t_::allowDriverInOrderLists(bool OnlyIfRequested) {
+  // Use in-order lists implementation from L0 driver instead
+  // of adapter's implementation.
+
+  // The following driver version is known to be passing and only this or newer
+  // drivers should be allowed by default for in order lists.
+#define L0_DRIVER_INORDER_MINOR_VERSION 6
+#define L0_DRIVER_INORDER_PATCH_VERSION 32149
+
+  static const bool UseDriverInOrderLists = [&] {
+    const char *UrRet = std::getenv("UR_L0_USE_DRIVER_INORDER_LISTS");
+    bool CompatibleDriver = this->isDriverVersionNewerOrSimilar(
+        1, L0_DRIVER_INORDER_MINOR_VERSION, L0_DRIVER_INORDER_PATCH_VERSION);
+    bool DriverInOrderRequested = UrRet ? std::atoi(UrRet) != 0 : false;
+    if (OnlyIfRequested) {
+      return DriverInOrderRequested;
+    }
+    bool CanUseDriverInOrderLists = CompatibleDriver || DriverInOrderRequested;
+    return CanUseDriverInOrderLists;
+  }();
+
+  return UseDriverInOrderLists;
```

Comment:
Hey @winstonzhang-intel , I realized that the OnlyIfRequested does not work as I had originally intended. that si why my removal before did not prevent the graph failures. Add an if condition 

```suggestion
  if (OnlyIfRequested) {
    const char *UrRet = std::getenv("UR_L0_USE_DRIVER_INORDER_LISTS");
    bool DriverInOrderRequested = UrRet ? std::atoi(UrRet) != 0 : false;
    return DriverInOrderRequested
  }
  return UseDriverInOrderLists;
```

Remove the:

    if (OnlyIfRequested) {
      return DriverInOrderRequested;
    }
    
    
  In the static lambda function.

Reply:
I see, this should fix the graph failures if we have the in order lists disabled.